### ISSUE TITLE
avoid linking to hdf5_hl

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#145]
i think the issue is that cmake's FindHDF5 includes this in HDF5_LIBRARIES, so we may want to try HDF5_C_LIBRARIES

**Comment by cyrush** [comment]
```
[harrison37@naples thirdparty_shared (master)]$ visit/hdf5/1.8.14/darwin-x86_64/bin/h5cc  -show
clang -I/Users/harrison37/Work/masonry/build-mb-trunk-darwin-10.11-x86_64/thirdparty_shared/visit/szip/2.1/darwin-x86_64/include -fno-common -fexceptions -O2 -L/Users/harrison37/Work/masonry/build-mb-trunk-darwin-10.11-x86_64/thirdparty_shared/visit/hdf5/1.8.14/darwin-x86_64/lib -lhdf5_hl -lhdf5 -L/Users/harrison37/Work/masonry/build-mb-trunk-darwin-10.11-x86_64/thirdparty_shared/visit/hdf5/1.8.14/darwin-x86_64/lib -L/Users/harrison37/Work/masonry/build-mb-trunk-darwin-10.11-x86_64/thirdparty_shared/visit/szip/2.1/darwin-x86_64/lib -lsz -lz -ldl -lm
[harrison37@naples thirdparty_shared (master)]$ ~/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/hdf5-1.8.17-pzxksxsggtm6ab72nbymwokfybp66boh/bin/h5cc -show
/Users/harrison37/Work/conduit/uberenv_libs/spack/lib/spack/env/clang/clang -I/Users/harrison37/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/szip-2.1-r5zci7g633gw6pbqeqjid5rvtvn5yb7r/include -L/Users/harrison37/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/hdf5-1.8.17-pzxksxsggtm6ab72nbymwokfybp66boh/lib /Users/harrison37/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/hdf5-1.8.17-pzxksxsggtm6ab72nbymwokfybp66boh/lib/libhdf5_hl.a /Users/harrison37/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/hdf5-1.8.17-pzxksxsggtm6ab72nbymwokfybp66boh/lib/libhdf5.a -L/Users/harrison37/Work/conduit/uberenv_libs/spack/opt/spack/darwin-x86_64/clang-3.4svn/szip-2.1-r5zci7g633gw6pbqeqjid5rvtvn5yb7r/lib -lsz -lz -ldl -lm
```

**Comment by cyrush** [comment]
Tried several things (only requesting C hdf5 libs, etc), but HL still crept in.

Added logic to filter it out from HDF5_LIBRARIES.

resolved in #144
